jvb/src/main/java/org/jitsi/videobridge/cc/BitrateController.java

```diff
@@ -94,7 +96,7 @@
  * @author George Politis
  */
 @SuppressWarnings("JavadocReference")
-public class BitrateController
+public class BitrateController<T extends BitrateController.MediaSourceContainer>
```

Comment:
Do we need to be generic around this?  Could it just always use the `MediaSourceContainer` type, rather than tying it to some specific type `T`?  I thought this might be there because we need to get the exact type `T` _out_, but I didn't see that (though I may have missed it).  I also see some places that do just use `MediaSourceContainer` instead of `T` (not just the static ones, but members as well I think?)

Reply:
The reason for the generics is to allow `Endpoint` to use `conference::getEndpoints` directly. It returns a `List<AbstractEndpoint>` and we need a `List<MediaSourceContainer>`.

Reply:
Ok, I get that...but the mixing and matching of `List<T>` and `List<MediaSourceContainer>` feels a bit weird, could it all use `T`?

Reply:
Yes, but it gets hairy. `EndpointMultiRank` becomes generic, so `makeEndpointMultiRankList()` can no longer be static...

Reply:
Actually, it can be static and generic. I'll push that change.